### DISCUSSION TITLE
In memory node error sending request for url (http://127.0.0.1:8011/)

### DISCUSSION THREAD

**Discussion by fbozhkov** [zkSync-Community-Hub/zksync-developers#815] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I am having issues when trying to deploy a smart contract. I set the rpc as such 
`--rpc-url http:/127.0.0.1:8011 `
and I get
```
[⠊] Compiling...
No files changed, compilation skipped
Error: error sending request for url (http://127.0.0.1:8011/)

Context:
- Error #0: error sending request for url (http://127.0.0.1:8011/)
- Error #1: client error (SendRequest)
- Error #2: connection closed before message completed
 ```
 
I have installed zksync cli and ran 
`npx zksync-cli dev config` setup for in memory node
`npx zksync-cli dev start`

I have verified that image is running on port 8011.

What am I missing?

**Comment by alhenaai** [comment]
It seems like you are encountering a connection issue with the in-memory node on port 8011. Here are a few steps you can take to troubleshoot this problem:

1. **Verify Docker Container**: Ensure that the Docker container for the in-memory node is running and properly configured to expose port 8011. You can check this by running `docker ps` to see if the container is up and the ports are correctly mapped.

2. **Check Network Configuration**: Sometimes, network configurations might prevent connections even if everything is set up correctly. Ensure there are no firewalls or network settings blocking the connection to `http://127.0.0.1:8011`.

3. **Reset MetaMask Cache**: If you are using MetaMask or a similar wallet, try resetting the cached data. In MetaMask, go to 'Settings' > 'Advanced' and select 'Reset Account' to clear the transaction history, which can sometimes interfere with new transactions.

4. **Debug Mode**: Try running the in-memory node in debug mode to get more detailed logs that might help identify the issue. You can start the node in debug mode by adding the `--debug` flag:
   ```
   npx zksync-cli dev start --debug
   ```

5. **Check Logs**: Look at the logs of the Docker container to see if there are any errors or warnings that could give more insight into what might be going wrong.

If after these steps you are still facing issues, it might be helpful to revisit the setup instructions for the in-memory node to ensure all steps were followed correctly. You can find detailed setup instructions and troubleshooting tips in the [ZKsync documentation on setting up and using the In-Memory Node](https://docs.zksync.io/build/test-and-debug/in-memory-node).

For further assistance, you might want to engage with the ZKsync developer community on [GitHub discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth support.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment, marked as answer]
It's something we are aware of and will be working on resolving. You can track the progress here: https://github.com/matter-labs/zksync-cli/issues/179

**Reply by bxpana** [reply]
In the mean time you can install the era test node instead and it should work https://docs.zksync.io/build/test-and-debug/in-memory-node#install-and-set-up-era_test_node
